File: ruby/test_measure.rb
require '/home/julien/Software/Cpp_Swig_Ruby_Python_MCVE/build-modif/Products/ruby/mylib.so'

class RubyTestMeasure < Mylib::Measure
  def name
    return "RubyTestMeasure"
  end

  def run_impl(runner)
    m = runner.get_current_model()
    puts "Ruby Model named: #{m.getName()}"
    m.setName("Ruby Model")
    m.pushOp("A Ruby Op")

    puts "Ruby: Model has: #{m.numObjects()} spaces"
    m.pushObject("Ruby Space")
    m.numObjects().times do |i|
      puts "* #{i} = #{m.getObject(i).getName()}"
    end

    # This crashes
    m.getObject(0).setName("MODIFIED FROM RUBY")
    return true;
  end
end

File: python/test_measure.py
# import sys
# sys.path.append("/home/julien/Software/Cpp_Swig_Ruby_Python_MCVE/build-modif/Products/python/")

from time import time, ctime
print('Today is',ctime(time()))

import mylib

class PythonTestMeasure(mylib.PythonMeasure):
    def __init__(self):
        mylib.PythonMeasure.__init__(self)
        print("Created Object (Python __init__)")

    def run_impl(self, r: mylib.Runner):
        m = r.get_current_model()
        print(f"Python Model named: {m.getName()}")
        m.pushOp("Op from Python")
        m.setName("Python Model")
        print(f"Python: {m.numObjects()} objects")
        m.pushObject("Python Space")

        for i in range(m.numObjects()):
            print(f" * {i} = {m.getObject(i).getName()}")

        m.getObject(1).setName("MODIFIED FROM PYTHON")

        return True
    def name(self):
        return "Python Test Measure"


def make_measure():
    return PythonTestMeasure()

